src/shared/ParamType.ts

```diff
@@ -0,0 +1,167 @@
+import {ValidationStatus} from './ValidationStatus'
+import p5 from 'p5'
+/**
+ * `ParamType` is the enum of all supported parameter types for visualizers
+ * and sequences. These types are supported by the interface and internal
+ * type checker
+ */
+export enum ParamType {
+    BOOLEAN,
+    COLOR,
+    NUMBER,
+    INTEGER,
+    BIGINT,
+    ENUM,
+    STRING,
+    NUMBER_ARRAY,
+    BIGINT_ARRAY,
+    VECTOR,
+}
```

Comment:
Yes these can certainly be documented. However, the reason that VECTOR and NUMBER_ARRAY are separate is because their realized types are unrelated. NUMBER_ARRAYs are realized as `number[]`, while VECTORs are realized as `p5.Vector`, so just having a NUMBER_ARRAY with length property wouldn't really make sense and would be more obfuscated than the current solution if certain lengths were to resolve to different types

Reply:
Got it. Indeed, that was not clear. The comments before each param type now at least make it clear in context here; presumably in a later PR this will be covered in the visualizer-writer manual (in fact, these comments can be made into markdown comments so they can be automatically extracted into said manual with mkdocs). Resolving this conversation.